packages/wallets/src/utils/options.ts

```diff
@@ -0,0 +1,18 @@
+import type { EVMSmartWallet, EVMMPCWallet, TransactionInput as EVMTransaction } from "@/evm";
+import type { SolanaMPCWallet, SolanaSmartWallet, Transaction as SolanaTransaction } from "@/solana";
+
+export type Wallet = EVMSmartWallet | EVMMPCWallet | SolanaSmartWallet | SolanaMPCWallet;
+export type Transaction = EVMTransaction | SolanaTransaction;
+
+export interface Callbacks {
+    onWalletCreationStart?: () => Promise<void>;
+    onWalletCreationComplete?: (wallet: Wallet) => Promise<void>;
+    onWalletCreationFail?: (error: Error) => Promise<void>;
+    onTransactionStart?: (transaction: Transaction) => Promise<void>;
+    onTransactionComplete?: (transaction: Transaction) => Promise<void>;
+    onTransactionFail?: (error: Error) => Promise<void>;
+}
+
+export interface WalletOptions {
+    experimental_callbacks?: Callbacks;
```

Comment:
i dont think these should be experimental as we already had them

Reply:
I don't think callbacks in their current form have a valid use case outside of React SDK. I also think we could modify the React SDK to not rely on callbacks at all while keeping all the functionality. Finally, if we decide to keep them, we might remove some, or change the parameters we pass.

Reply:
It's not required as we're doing `0.x` release and can do breaking changes, but IMO it's a good DX to communicate we might remove/change that feature specifically!

Reply:
If you have any suggestions on how to handle this without callbacks in the React SDK, would love to hear them!

Reply:
and that wrapper would live in the react sdk?

Reply:
Yup

Reply:
That could work. I still feel callbacks are cleaner and easier to integrate if anybody else ends up needing them